pkg/graph/resolvers/key_contact_test.go

```diff
@@ -0,0 +1,285 @@
+package resolvers
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+
+	"github.com/cms-enterprise/mint-app/pkg/email"
+	"github.com/cms-enterprise/mint-app/pkg/models"
+	"github.com/cms-enterprise/mint-app/pkg/storage/loaders"
+	"github.com/cms-enterprise/mint-app/pkg/userhelpers"
+)
+
+// TestCreateKeyContactUser validates creating a user contact for key contact.
+func (suite *ResolverSuite) TestCreateKeyContactUser() {
+	categoryName := "Test Category"
+	newCategory, err := CreateKeyContactCategory(
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		categoryName,
+	)
+	suite.NoError(err)
+
+	userName := "John Doe"
+	subjectArea := "Healthcare"
+	subjectCategoryID := newCategory.ID
+
+	contact, err := CreateKeyContactUser(
+		suite.testConfigs.Context,
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		nil,
+		nil,
+		email.AddressBook{},
+		userName,
+		subjectArea,
+		subjectCategoryID,
+		userhelpers.GetUserInfoAccountInfoWrapperFunc(suite.stubFetchUserInfo),
+	)
+	suite.NoError(err)
+	suite.NotNil(contact)
+	suite.Equal(subjectArea, contact.SubjectArea)
+	suite.Equal(subjectCategoryID, contact.SubjectCategoryID)
+}
+
+// TestCreateKeyContactMailbox validates creating a mailbox contact for key contact.
+func (suite *ResolverSuite) TestCreateKeyContactMailbox() {
+	categoryName := "Test Category"
+	newCategory, err := CreateKeyContactCategory(
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		categoryName,
+	)
+	suite.NoError(err)
+
+	mailboxTitle := "Support Team"
+	mailboxAddress := "support@example.com"
+	subjectArea := "Healthcare"
+	subjectCategoryID := newCategory.ID
+
+	contact, err := CreateKeyContactMailbox(
+		suite.testConfigs.Context,
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		nil,
+		nil,
+		email.AddressBook{},
+		mailboxTitle,
+		mailboxAddress,
+		subjectArea,
+		subjectCategoryID,
+	)
+
+	suite.NoError(err)
+	suite.NotNil(contact)
+	suite.Equal(mailboxTitle, *contact.Name)
```

Comment:
When dereferencing a pointer for a test, this can be a good practice (so it doesn't create a panic if nil). It can be confusing to troubleshoot if you don't do this. I'd recommend doing this in the other spots where you dereference a pointer
```suggestion
	if suite.NotNil(contact.Name){
		suite.Equal(mailboxTitle, *contact.Name)
	}
```